Sources/Widgets/Widgets3D/ResliceCursorWidget/index.js

```diff
@@ -0,0 +1,562 @@
+    vec4.transformMat4(newOriginXYZW, originXYZW, newResliceAxes);
+
+    newResliceAxes[4 * 3 + 0] = newOriginXYZW[0];
+    newResliceAxes[4 * 3 + 1] = newOriginXYZW[1];
+    newResliceAxes[4 * 3 + 2] = newOriginXYZW[2];
+
+    // Compute a new set of resliced extents
+    let extentX = 0;
+    let extentY = 0;
+
+    // Pad extent up to a power of two for efficient texture mapping
+    // make sure we're working with valid values
+    const realExtentX =
+      spacingX === 0 ? Number.MAX_SAFE_INTEGER : planeSizeX / spacingX;
+
+    // Sanity check the input data:
+    // * if realExtentX is too large, extentX will wrap
+    // * if spacingX is 0, things will blow up.
+
+    const value = VTK_INT_MAX >> 1; // eslint-disable-line no-bitwise
+
+    if (realExtentX > value) {
+      vtkErrorMacro(
+        'Invalid X extent: ',
+        realExtentX,
+        ' on view type : ',
+        viewType
+      );
+      extentX = 0;
+    } else {
+      extentX = 1;
+      while (extentX < realExtentX) {
+        extentX <<= 1; // eslint-disable-line no-bitwise
+      }
+    }
+
+    // make sure extentY doesn't wrap during padding
+    const realExtentY =
+      spacingY === 0 ? Number.MAX_SAFE_INTEGER : planeSizeY / spacingY;
+
+    if (realExtentY > value) {
+      vtkErrorMacro(
+        'Invalid Y extent:',
+        realExtentY,
+        ' on view type : ',
+        viewType
+      );
+      extentY = 0;
+    } else {
+      extentY = 1;
+      while (extentY < realExtentY) {
+        extentY <<= 1; // eslint-disable-line no-bitwise
+      }
+    }
+
+    const outputSpacingX = extentX === 0 ? 1.0 : planeSizeX / extentX;
+    const outputSpacingY = extentY === 0 ? 1.0 : planeSizeY / extentY;
+
+    // let modify = false;
+
+    // for (let i = 0; i < 4; i++) {
+    //   for (let j = 0; j < 4; j++) {
+    //     const index = 4 * j + i;
+
+    //     const d = newResliceAxes[index];
+
+    //     if (d !== model.resliceAxes[index]) {
+    //       model.resliceAxes[index] = d;
+    //       modify = true;
+    //     }
+    //   }
+    // }
+
+    // if (modify) {
+    //   publicAPI.modified();
+
+    //   publicAPI.setResliceParameters(
+    //     outputSpacingX,
+    //     outputSpacingY,
+    //     extentX,
+    //     extentY
+    //   );
+    // }
+
+    // console.log(newResliceAxes);
+
+    imageReslice.setResliceAxes(newResliceAxes);
+    imageReslice.setOutputSpacing([outputSpacingX, outputSpacingY, 1]);
+    imageReslice.setOutputOrigin([
+      0.5 * outputSpacingX,
+      0.5 * outputSpacingY,
+      0,
+    ]);
+    imageReslice.setOutputExtent([0, extentX - 1, 0, extentY - 1, 0, 0]);
+
+    return newResliceAxes;
+  };
+
+  publicAPI.getPlaneAndPlaneSourceFromViewType = (type) => {
+    const returnedPlaneSource = vtkPlaneSource.newInstance();
+    const returnedPlane = vtkPlane.newInstance();
+    const widgetState = publicAPI.getWidgetState();
+    const origin = widgetState.getCenter();
+    returnedPlaneSource.setOrigin(origin);
```

Comment:
the origin of a planesource should be the lowerleft corner, not necessarily the intersection point of all planes.

Reply:
This method is used to recompute the right three points that define the plane. We only need the normal and one point on the plane.

Reply:
For the record, I think it might lead to bugs later on, but it is fine for now